lib/flop/builder.ex

```diff
@@ -120,12 +120,21 @@ defmodule Flop.Builder do
   for op <- @operators do
     {fragment, prelude, combinator} = op_config(op)
 
-    defp build_op(c, _schema_struct, {:normal, field}, %Filter{
-           op: unquote(op),
-           value: value
-         }) do
+    defp build_op(
+           c,
+           schema_struct,
+           {:normal, field},
+           %Filter{
+             op: unquote(op),
+             value: value
+           } = filter
+         ) do
       unquote(prelude)
-      build_dynamic(unquote(fragment), false, unquote(combinator))
+
+      case runtime_dynamic(c, schema_struct, field, filter) do
+        nil -> build_dynamic(unquote(fragment), false, unquote(combinator))
+        dynamic -> dynamic
+      end
```

Comment:
I think you could add another `build_op` function clause before the `for op <- @operators do` and remove `:not_empty` and `:empty` from the `@operators`. Then you don't need this extra `case`.

Reply:
Unfortunately I was not able to do that as `fragment` is not available "in the right form" in the other context. Probably bc of my lack of macro understanding.

Reply:
Alright, let's leave it as it is then.